### ISSUE TITLE
risky-octal validation partially broken for non-quoted permissions

### ISSUE THREAD

**Issue by tomsiewert** [ansible/ansible-lint#2758]
##### Summary

If the file permission is set to `0646`, ansible-lint throws a `risky-octal` error. When the file permission is quoted or set to something else than `0646`, the issue does not occur.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.14.0]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.11/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.11.0 (main, Nov 12 2022, 07:10:51) [GCC 11.2.1 20220219] (/usr/local/bin/python)
  jinja version = 3.1.2
  libyaml = False

ansible-lint --version
ansible-lint 6.9.0 using ansible 2.14.0

```

- ansible installation method: pip3
- ansible-lint installation method: pip3

##### OS / ENVIRONMENT

uname: `6.0.9 #1-NixOS SMP PREEMPT_DYNAMIC Wed Nov 16 09:04:15 UTC 2022 x86_64 GNU/Linux`

ansible-lint runs in a `python:alpine` Docker image

##### STEPS TO REPRODUCE

ansible-lint config:
```
--- .ansible-lint ---
# merge-with-defaults
skip_list:
  - package-latest
  - no-changed-when
  - schema
  - experimental
  - name[play]
  - name[casing]
  - name[template]
  - jinja[spacing]
  - template-instead-of-copy
warn_list:
  - jinja[invalid]
exclude_paths:
  - .git/
  - .cache/
  - roles/
  - files/
enable_list:
  - yaml
extra_vars:
  servers: localhost
  hostvars:
    DUMMY_HOST:
      hosts:
        - localhost
kinds:
  - playbook: '**/play*.{yml,yaml}'
---------------

--- .yamllint ---
# merge-with-defaults
rules:
  brackets:
    min-spaces-inside: 1
    max-spaces-inside: 1
    min-spaces-inside-empty: 0
    max-spaces-inside-empty: 0
    level: error
  braces:
    max-spaces-inside: 1
    min-spaces-inside: 1
    min-spaces-inside-empty: 0
    max-spaces-inside-empty: 0
    level: error
  line-length:
    max: 160
    ignore: |
      /group_vars/
      /host_vars/
---------------

```

Broken check:
```yaml
---
- hosts: localhost
  gather_facts: false
  tasks:
    - name: Change permission of test.log
      ansible.builtin.file:
        path: /var/log/test.log
        mode: 0646
```

Working check:
```yaml
---
- hosts: localhost
  gather_facts: false
  tasks:
    - name: Change permission of test.log
      ansible.builtin.file:
        path: /var/log/test.log
        mode: '0646'

---
- hosts: localhost
  gather_facts: false
  tasks:
    - name: Change permission of test.log
      ansible.builtin.file:
        path: /var/log/test.log
        mode: 0644
```

##### Desired Behavior

ansible-lint passes the check.

##### Actual Behavior

ansible-lint fails with `risky-octal` error.

```
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/root/.cache/ansible-compat/77f31b', colored=True, configured=True, cwd=PosixPath('/ansible-repo'), display_relative_path=True, exclude_paths=['/ansible-repo/.git', '/ansible-repo/.cache', '/ansible-repo/roles', '/ansible-repo/files'], format='rich', lintables=['play_test.yml'], list_rules=False, list_tags=False, write_list=[], parseable=False, quiet=False, rulesdirs=[PosixPath('/ansible-lint/custom_rules'), '/usr/local/lib/python3.11/site-packages/ansiblelint/rules'], skip_list=['package-latest', 'no-changed-when', 'schema', 'experimental', 'name[play]', 'name[casing]', 'name[template]', 'jinja[spacing]', 'template-instead-of-copy'], tags=[], verbosity=17, warn_list=['jinja[invalid]'], kinds=[{'playbook': '**/play*.{yml,yaml}'}, {'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.{yaml,yml}'}, {'requirements': '**/meta/requirements.{yaml,yml}'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'arg_specs': '**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.{yaml,yml}'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], mock_filters=[], mock_modules=[], mock_roles=[], loop_var_prefix=None, only_builtins_allow_collections=[], var_naming_pattern=None, offline=False, project_dir='.', extra_vars={'servers': 'localhost', 'hostvars': {'DUMMY_HOST': {'hosts': ['localhost']}}}, enable_list=['yaml'], skip_action_validation=True, strict=False, rules={}, profile=None, list_profiles=False, progressive=False, rulesdir=[PosixPath('/ansible-lint/custom_rules')], use_default_rules=True, config_file='/ansible-repo/.ansible-lint', version=False, cache_dir_lock=<filelock._unix.UnixFileLock object at 0x7f063fade190>)
DEBUG    /ansible-repo
DEBUG    Loading custom .yamllint config file, this extends our internal yamllint config.
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 1, 'max-spaces-inside': 1, 'min-spaces-inside-empty': 0, 'max-spaces-inside-empty': 0}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 1, 'max-spaces-inside': 1, 'min-spaces-inside-empty': 0, 'max-spaces-inside-empty': 0}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'float-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False, 'ignore': <pathspec.pathspec.PathSpec object at 0x7f063e959f50>}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
INFO     Set ANSIBLE_LIBRARY=/root/.cache/ansible-compat/77f31b/modules:/root/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/root/.cache/ansible-compat/77f31b/collections:/root/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/root/.cache/ansible-compat/77f31b/roles:roles:/root/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Executing syntax check on play_test.yml (0.88s)
DEBUG    Examining play_test.yml of type playbook
WARNING  Listing 1 violation(s) that are fatal
risky-octal: Octal file permissions must contain leading zero or be a string.
play_test.yml:5 Task/Handler: Change permission of test.log

DEBUG    Attempting to release lock 139664814891408 on /root/.cache/ansible-compat/77f31b/.lock
DEBUG    Lock 139664814891408 released on /root/.cache/ansible-compat/77f31b/.lock
You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - risky-octal  # Octal file permissions must contain leading zero or be a string.

DEBUG    Determined rule-profile order: {'internal-error': (0, 'min'), 'load-failure': (1, 'min'), 'parser-error': (2, 'min'), 'syntax-check': (3, 'min'), 'command-instead-of-module': (4, 'basic'), 'command-instead-of-shell': (5, 'basic'), 'deprecated-bare-vars': (6, 'basic'), 'deprecated-command-syntax': (7, 'basic'), 'deprecated-local-action': (8, 'basic'), 'deprecated-module': (9, 'basic'), 'inline-env-var': (10, 'basic'), 'key-order': (11, 'basic'), 'literal-compare': (12, 'basic'), 'jinja': (13, 'basic'), 'no-jinja-when': (14, 'basic'), 'no-tabs': (15, 'basic'), 'partial-become': (16, 'basic'), 'playbook-extension': (17, 'basic'), 'role-name': (18, 'basic'), 'schema': (19, 'basic'), 'name': (20, 'basic'), 'var-naming': (21, 'basic'), 'yaml': (22, 'basic'), 'name[template]': (23, 'moderate'), 'name[imperative]': (24, 'moderate'), 'name[casing]': (25, 'moderate'), 'no-free-form': (26, 'moderate'), 'spell-var-name': (27, 'moderate'), 'avoid-implicit': (28, 'safety'), 'latest': (29, 'safety'), 'package-latest': (30, 'safety'), 'risky-file-permissions': (31, 'safety'), 'risky-octal': (32, 'safety'), 'risky-shell-pipe': (33, 'safety'), 'galaxy': (34, 'shared'), 'ignore-errors': (35, 'shared'), 'layout': (36, 'shared'), 'meta-incorrect': (37, 'shared'), 'meta-no-info': (38, 'shared'), 'meta-no-tags': (39, 'shared'), 'meta-video-links': (40, 'shared'), 'meta-version': (41, 'shared'), 'meta-unsupported-ansible': (42, 'shared'), 'no-changed-when': (43, 'shared'), 'no-changelog': (44, 'shared'), 'no-handler': (45, 'shared'), 'no-relative-paths': (46, 'shared'), 'max-block-depth': (47, 'shared'), 'max-tasks': (48, 'shared'), 'unsafe-loop': (49, 'shared'), 'avoid-dot-notation': (50, 'production'), 'disallowed-ignore': (51, 'production'), 'fqcn': (52, 'production'), 'import-task-no-when': (53, 'production'), 'meta-no-dependencies': (54, 'production'), 'single-entry-point': (55, 'production'), 'use-loop': (56, 'production')}
             Rule Violation Summary             
 count tag         profile rule associated tags 
     1 risky-octal safety  formatting           

Failed after moderate profile, 2/5 star rating: 1 failure(s), 0 warning(s) on 1 files.
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
In this case that is not really a bug, more of a feature. Still, we decided to improve the documentation and the messaging in order to clarify what is expected from the user.

Shortly, don't use numeric values because we cannot distinguish between a decimal and an octal, use strings as the values of these are more predictable. Feel free to review the pending pull request, especially the new error message as we are trying to make it clear while keeping its size under control.
